### ISSUE TITLE
Add support for basic auth

### ISSUE THREAD

**Issue by spiffzilla** [misiupajor/check_elasticsearch#2]
When Elasticsearch are behind a basic auth login the plugin wont work.
Have tried do use plugin with a ES behind an Nginx proxy.

**Comment by misiupajor** [comment]
Thanks for the input @spiffzilla and you are indeed correct that it will not work as intended when basic auth is available. There is fundamental functionalities for supporting basic auth within the plugin, so I consider this an easy fix. I'll implement during the day and revert back.

**Comment by spiffzilla** [comment]
Great, I assume it will be argument controlled behaviour of the http auth functionality
.

**Comment by misiupajor** [comment]
Yes I think that would be the best approach, but I will make the arguments optional. If they are defined then basic auth is assumed

**Comment by misiupajor** [comment]
This is taking a bit longer than expected, but it's in the works. I'll ping you when I've fixed this!
